File: tea-app-android/lib/models/order.dart
class Order {
  final int? id;
  final String orderNo;
  final int userId;
  final int staffId;
  final int customProductId;
  final Map<String, dynamic>? customProductSnapshot;
  final String state;
  final double unitPrice;
  final int quantity;
  final double shippingCost;
  final double totalAmount;
  final String hsCode;
  final String countryOfOrigin;
  final Map<String, dynamic>? billingAddressSnapshot;
  final Map<String, dynamic>? deliveryAddressSnapshot;
  final int? liveRoomId;
  final String? createdAt;

  Order({
    this.id, required this.orderNo, required this.userId, required this.staffId,
    required this.customProductId, this.customProductSnapshot,
    required this.state, required this.unitPrice, required this.quantity,
    required this.shippingCost, required this.totalAmount,
    required this.hsCode, required this.countryOfOrigin,
    this.billingAddressSnapshot, this.deliveryAddressSnapshot,
    this.liveRoomId, this.createdAt,
  });

  factory Order.fromJson(Map<String, dynamic> j) => Order(
    id: j['id'], orderNo: j['order_no'] ?? '', userId: j['user_id'] ?? 0,
    staffId: j['staff_id'] ?? 0, customProductId: j['custom_product_id'] ?? 0,
    customProductSnapshot: j['custom_product_snapshot'], state: j['state'] ?? '',
    unitPrice: (j['unit_price'] ?? 0).toDouble(), quantity: j['quantity'] ?? 1,
    shippingCost: (j['shipping_cost'] ?? 0).toDouble(), totalAmount: (j['total_amount'] ?? 0).toDouble(),
    hsCode: j['hs_code'] ?? '0902.20', countryOfOrigin: j['country_of_origin'] ?? 'China',
    billingAddressSnapshot: j['billing_address_snapshot'],
    deliveryAddressSnapshot: j['delivery_address_snapshot'],
    liveRoomId: j['live_room_id'], createdAt: j['created_at'],
  );
}

File: tea-system/internal/models/order.go
package models

import (
	"time"
)

// Order states
const (
	OrderStateOrdering            = "ordering"
	OrderStatePaid                = "paid"
	OrderStatePendingDeclaration  = "pending_declaration"
	OrderStateProducing           = "producing"
	OrderStateReadyForProduction  = "ready_for_production"
	OrderStatePendingCustoms      = "pending_customs"
	OrderStateCustomsClear        = "customs_clear"
	OrderStateShipped             = "shipped"
	OrderStateCompleted           = "completed"
	OrderStateCancelled           = "cancelled"
	OrderStateDisputed            = "disputed"
)

// Order — 订单表
type Order struct {
	ID                    uint64    `gorm:"primaryKey;column:id" json:"id"`
	OrderNo               string    `gorm:"column:order_no;uniqueIndex;not null;size:30" json:"order_no"`
	UserID                *uint64   `gorm:"column:user_id;index" json:"user_id"`
	StaffID               uint64    `gorm:"column:staff_id;not null" json:"staff_id"`
	CustomProductID       uint64    `gorm:"column:custom_product_id;not null" json:"custom_product_id"`
	CustomProductSnapshot JSONMap   `gorm:"column:custom_product_snapshot;not null;type:jsonb" json:"custom_product_snapshot"`
	State                 string    `gorm:"column:state;not null;size:30" json:"state"`
	UnitPrice             float64   `gorm:"column:unit_price;not null;type:numeric(10,2)" json:"unit_price"`
	Quantity              int       `gorm:"column:quantity;not null" json:"quantity"`
	ShippingCost          float64   `gorm:"column:shipping_cost;not null;type:numeric(10,2)" json:"shipping_cost"`
	TotalAmount           float64   `gorm:"column:total_amount;not null;type:numeric(10,2)" json:"total_amount"`
	HsCode                string    `gorm:"column:hs_code;not null;size:20" json:"hs_code"`
	CountryOfOrigin       string    `gorm:"column:country_of_origin;not null;size:50" json:"country_of_origin"`
	BillingAddressSnapshot  JSONMap  `gorm:"column:billing_address_snapshot;not null;type:jsonb" json:"billing_address_snapshot"`
	DeliveryAddressSnapshot JSONMap  `gorm:"column:delivery_address_snapshot;not null;type:jsonb" json:"delivery_address_snapshot"`
	LiveRoomID            *uint64   `gorm:"column:live_room_id" json:"live_room_id,omitempty"`
	TrackingNumber        string    `gorm:"column:tracking_number;size:60" json:"tracking_number,omitempty"`
	ShippingCarrier       string    `gorm:"column:shipping_carrier;size:50" json:"shipping_carrier,omitempty"`
	ShippedAt             *time.Time `gorm:"column:shipped_at" json:"shipped_at,omitempty"`
	CreatedAt             time.Time `gorm:"column:created_at;not null" json:"created_at"`
	UpdatedAt             time.Time `gorm:"column:updated_at;not null" json:"updated_at"`
	DeletedAt             *time.Time `gorm:"column:deleted_at" json:"-"`

	User          *User         `gorm:"-:migration;foreignKey:UserID" json:"user,omitempty"`
	Staff         *Staff        `gorm:"-:migration;foreignKey:StaffID" json:"staff,omitempty"`
	CustomProduct *CustomProduct `gorm:"-:migration;foreignKey:CustomProductID" json:"custom_product,omitempty"`
	LiveRoom      *LiveRoom     `gorm:"-:migration;foreignKey:LiveRoomID" json:"live_room,omitempty"`
}

func (Order) TableName() string {
	return "orders"
}
